### PR TITLE
upgrade to AGP7

### PR DESCRIPTION
**Pull request by rosariopfernandes**
See #1966 

This PR should:
- [x] update AGP to 7.0.0 stable
- [x] update Gradle to 7.0.2
- [x] update the GitHub Actions workflow to use Java 11 - this is required by Gradle 7.
- [x] remove unused strings, colors and ids to avoid `UnusedIds` and `UnusedResources` lint errors.
- [x] suppress the `UnusedIds` lint error thrown in the `app` module - seems like AGP7 can't yet detect when an id is being used by ViewBinding.
- [x] suppress the `MediaCapabilities` lint error thrown in the `app` module - I couldn't figure out where this error is coming from, but I'm guessing Glide is responsible for it. 
- [x] fix many other lint issues that were thrown by AGP7

### CODE REVIEW

File: auth/src/main/res/values-ar/strings.xml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 <resources tools:ignore="ExtraTranslation" xmlns:tools="http://schemas.android.com/tools" xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     <string name="fui_progress_dialog_loading">جارٍ التحميل…</string>
     <string name="fui_sign_in_default">تسجيل الدخول</string>
-    <string name="fui_cancel">إلغاء</string>
```
Comment (samtstern) [inline auth/src/main/res/values-ar/strings.xml:4]: I am surprised we don't use "Cancel" anywhere!

File: database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java [hunk 2]
```diff
@@ -57,7 +57,6 @@ public void startListening() {
     @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
     public void stopListening() {
         mSnapshots.removeChangeEventListener(this);
-        notifyDataSetChanged();
```
Comment (samtstern) [inline database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java:60]: Is this change related to AGP7?

**Comment by samtstern** [comment]
@rosariopfernandes you are a legend! I can't believe you got this all working so quickly.

File: database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java (hunk 2 above)
Comment (rosariopfernandes) [reply database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java:60]: @samtstern Kind of. Actually, lint got mad at this method call. It suggests calling a more specific one such as `notifyItemInserted`, `notifyItemDeleted`, `notifyItemMoved` etc. 

But I figured that since we're only removing the listener and not adding/removing any item, maybe there's nothing to notify the adapter about?

**Comment by rosariopfernandes** [comment]
@samtstern I was just about to push that, OMG :laughing:

**Comment by samtstern** [comment]
@rosariopfernandes you were right about July 28th! 7.0 is out and I just updated your branch (I wanted to be faster than you for once in my life haha)

File: database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java (hunk 2 above)
Comment (samtstern) [reply database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java:60]: SGTM

**Comment by rosariopfernandes** [comment]
@samtstern You did it, I'm impressed. I bow down to you :laughing:

**Comment by samtstern** [comment]
I'll release 8.0 tomorrow!

**Comment by rosariopfernandes** [comment]
I was updating rtdb to use Paging 3, but there were some runtime errors, I
think I got something wrong.
I'll send a PR with the changes tomorrow so we can have a look at it
together.

**Comment by samtstern** [comment]
@rosariopfernandes oh wow I almost forgot about that change! Thanks for the reminder. No rush on that PR, 8.0 will go out when that's ready.
